pkg/controller/verticalpodautoscaler/verticalpodautoscaler_test.go

```diff
@@ -180,6 +180,48 @@ func TestOverrideArgs(t *testing.T) {
 
 }
 
+func TestOverrideNodeSelector(t *testing.T) {
+	vpa := NewVerticalPodAutoscaler()
+	r := newFakeReconciler(vpa, &appsv1.Deployment{})
+
+	selOverride := map[string]string{"node-role.kubernetes.io/infra": ""}
+
+	vpa.Spec.DeploymentOverrides.Admission.NodeSelector = selOverride
+	vpa.Spec.DeploymentOverrides.Recommender.NodeSelector = selOverride
+	vpa.Spec.DeploymentOverrides.Updater.NodeSelector = selOverride
+
+	for _, params := range controllerParams {
+		t.Run(fmt.Sprintf("override %s node selector", params.AppName), func(t *testing.T) {
+			podSpec := params.PodSpecMethod(r, vpa, params)
+			assert.Equal(t, selOverride, podSpec.NodeSelector)
+		})
+	}
+}
+
+func TestOverrideTolerations(t *testing.T) {
+	vpa := NewVerticalPodAutoscaler()
+	r := newFakeReconciler(vpa, &appsv1.Deployment{})
+
+	tolOverride := []corev1.Toleration{
+		{
+			Key:      "node-role.kubernetes.io/infra",
+			Effect:   corev1.TaintEffectNoSchedule,
+			Operator: corev1.TolerationOpExists,
+		},
+	}
+
+	vpa.Spec.DeploymentOverrides.Admission.Tolerations = tolOverride
+	vpa.Spec.DeploymentOverrides.Recommender.Tolerations = tolOverride
+	vpa.Spec.DeploymentOverrides.Updater.Tolerations = tolOverride
+
+	for _, params := range controllerParams {
+		t.Run(fmt.Sprintf("override %s node selector", params.AppName), func(t *testing.T) {
```

Comment:
oops, I copy/pasted the previous test and forgot to rename it. Should be "override %s tolerations"